# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RDDProcessManager.h
# Commit: e8fac91d4012
# Full Hash: e8fac91d4012b0ac5acce2d73847335d3828321b
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2021-01-04 09:08:57
# Regressor Bug: 1681043
# File Overlap Count: 2
# Description:
#   Bug 1681043 - P1. Launch RDD process early. r=alwu
#   
#   We can launch the RDD process early now that it is a fully asynchronous process and doesn't block anything.
#   
#   This will allow to retrieve the decoding capabilities of the RDD process right away.
# ==============================================================================

diff -r 5d453db46486 -r e8fac91d4012 dom/media/ipc/RDDProcessManager.h
--- a/dom/media/ipc/RDDProcessManager.h	Sun Jan 03 20:32:29 2021 +0000
+++ b/dom/media/ipc/RDDProcessManager.h	Sun Jan 03 23:20:38 2021 +0000
@@ -31,6 +31,8 @@
 
   using EnsureRDDPromise =
       MozPromise<ipc::Endpoint<PRemoteDecoderManagerChild>, nsresult, true>;
+  // Launch a new RDD process asynchronously
+  RefPtr<GenericNonExclusivePromise> LaunchRDDProcess();
   // If not using a RDD process, launch a new RDD process asynchronously and
   // create a RemoteDecoderManager bridge
   RefPtr<EnsureRDDPromise> EnsureRDDProcessAndCreateBridge(
@@ -104,6 +106,9 @@
   // the initial map is passed in command-line arguments) to be sent
   // when the process can receive IPC messages.
   nsTArray<dom::Pref> mQueuedPrefs;
+  // Promise will be resolved when the RDD process has been fully started and
+  // VideoBridge configured. Only accessed on the main thread.
+  RefPtr<GenericNonExclusivePromise> mLaunchRDDPromise;
 };
 
 }  // namespace mozilla
